src/libs/routes/authMiddleware.ts

```diff
@@ -33,4 +33,4 @@ export default (module, permissionType) => (req: Request, res: Response, next: N
      message: error.message
    });
  }
-}; 
\ No newline at end of file
+}; 
```

Comment:
1.  use destructuring in secretKey, role
2. verify in the case of no token 
3. give proper error messages saying "Unauthorized"
4. in the condition of hasPermission the message is not proper it should be "Permission denied"
5. pass the user in request user also 
6. call next function at the end

Reply:
Done

Reply:
1,2,5 is not done